lib/pure/memfiles.nim

```diff
@@ -245,6 +245,19 @@ proc open*(filename: string, mode: FileMode = fmRead,
       if close(result.handle) == 0:
         result.handle = -1
 
+proc flush*(f: var MemFile): bool =
+  when defined(windows):
+    result = flushViewOfFile(f.mem, f.size.int32) == 0
+    if not result:
+      let lastErr = osLastError()
+      if lastErr != ERROR_LOCK_VIOLATION.OSErrorCode:
```

Comment:
Why is this particular error ignored?

Reply:
[From MSDN](https://msdn.microsoft.com/en-us/library/windows/desktop/aa366563(v=vs.85).aspx):
<details><summary>The long remarks</summary>
Remarks

Flushing a range of a mapped view initiates writing of dirty pages within that range to the disk. Dirty pages are those whose contents have changed since the file view was mapped. The FlushViewOfFile function does not flush the file metadata, and it does not wait to return until the changes are flushed from the underlying hardware disk cache and physically written to disk. To flush all the dirty pages plus the metadata for the file and ensure that they are physically written to disk, call FlushViewOfFile and then call the FlushFileBuffers function.

When flushing a memory-mapped file over a network, FlushViewOfFile guarantees that the data has been written from the local computer, but not that the data resides on the remote computer. The server can cache the data on the remote side. Therefore, FlushViewOfFile can return before the data has been physically written to disk.
</details>

Reply:
Oh!
I found this code: [MappedByteBuffer.c#L66](https://github.com/houssemmh/openjdk8u/blob/fd9b4a042d1747a777d2181d01cfa38ced6ccaaa/jdk/src/windows/native/java/nio/MappedByteBuffer.c#L66)

```c
 retry = 0;
    do {
        result = FlushViewOfFile(a, (DWORD)len);
        if ((result != 0) || (GetLastError() != ERROR_LOCK_VIOLATION))
            break;
        retry++;
    } while (retry < 3);
``

Reply:
That code retries the flushing operation a maximum of 3 times. After 3 times, if the lock error occurs, it is still raised.

Reply:
I see.
Therefore I ignore ```ERROR_LOCK_VIOLATION```.

Reply:
Maybe add a similar code both for Windows and for Posix?